### PR TITLE
[ci] Use Github Action for uploading ReFrame releases on PyPI

### LINKED ISSUE #1559: Deploy new versions to PyPI using Github Actions

### PR DESCRIPTION
**Pull request by teojgo**
Fixes #1559

### CODE REVIEW

**Review by ekouts — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Deploy and Publish ReFrame to PyPI
+
+on:
+  release:
+    types: [released]
```
Comment (vkarak) [inline .github/workflows/publish.yml:5]: Does this cover pre-releases too?

**Review by vkarak — APPROVED** [review]
lgtm

File: .github/workflows/publish.yml (hunk 1 above)
Comment (teojgo) [reply .github/workflows/publish.yml:5]: No, the various alternatives are given here: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release

Comment (teojgo) [reply .github/workflows/publish.yml:5]: No, the alternatives types are given here: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release

Comment (vkarak) [reply .github/workflows/publish.yml:5]: You should then add `prereleased` here, too.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Review by vkarak — APPROVED** [review]
lgtm

**Pull request merged**